packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/query/QueryPredicateBuilder.kt

```diff
@@ -144,5 +144,14 @@ class QueryPredicateBuilder {
         fun fromSerializedMap(serializedMap: Map<String, Any>?): QueryPredicate? {
             return fromSerializedMap(serializedMap, null)
         }
+
+        @JvmStatic
+        fun extractIdFieldName(fieldName: String): String {
```

Comment:
Is this just to be backwards compatible with old codegen? The codegen fix alone should work, right?

If it's to be backwards compatible? Should we have a test which verifies this old behavior?

Reply:
Yes. This is for backwards compatibility.
I'll add a test to assert the original behavior.